scripts/automated_ingestion/eessi_task.py

```diff
@@ -0,0 +1,1201 @@
+                                   f"{next_state.name}\n",
+                                   commit_message,
+                                   branch_name=branch_name)
+        return result
+
+    @log_function_entry_exit()
+    def _init_payload_object(self):
+        """Initialize the payload object"""
+        if self.payload is not None:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO", "payload object already initialized")
+            return
+
+        # get name of of payload from metadata
+        payload_name = self.description.metadata["payload"]["filename"]
+        log_message(LoggingScope.TASK_OPS_DETAILS, "INFO", "payload_name: '%s'", payload_name)
+
+        # get config and remote_client from self.description.task_object
+        config = self.description.task_object.config
+        remote_client = self.description.task_object.remote_client
+
+        # determine remote_file_path by replacing basename of remote_file_path in self.description.task_object
+        #   with payload_name
+        description_remote_file_path = self.description.task_object.remote_file_path
+        payload_remote_file_path = os.path.join(os.path.dirname(description_remote_file_path), payload_name)
+        log_message(LoggingScope.TASK_OPS_DETAILS, "INFO", "payload_remote_file_path: '%s'", payload_remote_file_path)
+
+        # initialize payload object
+        payload_object = EESSIDataAndSignatureObject(config, payload_remote_file_path, remote_client)
+        self.payload = EESSITaskPayload(payload_object)
+        log_message(LoggingScope.TASK_OPS, "INFO", "payload: '%s'", self.payload)
+
+    @log_function_entry_exit()
+    def _handle_add_new_task(self):
+        """Handler for ADD action in NEW_TASK state"""
+        log_message(LoggingScope.TASK_OPS, "INFO", "Handling ADD action in NEW_TASK state: '%s'",
+                    self.description.get_task_file_name())
+        # determine next state
+        next_state = self._next_state(EESSITaskState.NEW_TASK)
+        log_message(LoggingScope.TASK_OPS_DETAILS, "INFO", "next_state: '%s'", next_state)
+
+        # initialize payload object
+        self._init_payload_object()
+
+        # update TaskState file content
+        self._update_task_state_file(next_state)
+
+        # TODO: verify that the sequence number is still valid (PR corresponding to the sequence number
+        #   is still open or yet to be created); if it is not valid, perform corrective actions
+        return next_state
+
+    @log_function_entry_exit()
+    def _find_pr_for_branch(self, branch_name: str) -> Optional[PullRequest]:
+        """
+        Find the single PR for the given branch in any state.
+
+        Args:
+            repo: GitHub repository
+            branch_name: Name of the branch
+
+        Returns:
+            PullRequest object if found, None otherwise
+        """
+        try:
+            prs = [pr for pr in list(self.git_repo.get_pulls(state="all"))
```

Comment:
Should this somehow be limited to PRs that also _target_ a certain branch (the default branch)?